utils/helper.js

```diff
@@ -217,3 +217,38 @@ function addBackgroundOverlay() {
         img.style.borderRadius = '15px 0 0 15px'
     }
 }
+
+// eslint-disable-next-line
+function setFullPageBackground(headingData, imagePath) {
+    const slideBackgrounds = document.getElementsByClassName('slide-background')
+    for (const [index, slideBackground] of Array.from(slideBackgrounds).entries()) {
+        const slide = slideBackground.classList.value
+            .replace('slide-background', '')
+            .replace(/(past|present|future)/, '')
+            .trim()
+        if (slide === 'image') {
+            // the slideNumber starts form 2 for presentation containing cover and toc, i.e. the first headingData refers to the third slide
+            // the slideNumber starts from 1 for presentation containing cover but no toc, i.e. the first headingData refers to the second slide
+            // so getting slide specific data from headingData by subtracting index value by first value of slideNumber
+            const slideData = headingData[index - headingData[0].slideNumber]
+            if (slideData.background && slideData.pdfbackground) {
+                slideBackground.style.backgroundImage = `url('${imagePath}/${slideData.background}')`
+                slideBackground.style.backgroundRepeat = 'no-repeat'
+                slideBackground.style.backgroundSize = 'cover'
+                slideBackground.style.backgroundPosition = 'center'
+
+                // update the background only when the window size changes to pdf size
+                // for normal browser window size, or if the browser is minimized do not update the image
+                const pdfWidth = 1123
+                const pdfHeight = 794
```

Comment:
these magic numbers are also defined in https://github.com/opf/revealjs-awesoMD-templates/blob/29c9cacb4a6ea8d2edb5b0a78c6269ce91f420d7/utils/exporter.js#L21
To prevent future bugs and typos it would be good to have them defined only in one place.